render/fx_renderer/gles2/shaders/corner_alpha.frag

```diff
@@ -1,21 +1,19 @@
-// TODO: surely I can be optimized
-float corner_alpha(vec2 size, vec2 position, float radius,
-            bool round_tl, bool round_tr, bool round_bl, bool round_br) {
+float get_dist(vec2 q, float radius) {
+	return min(max(q.x, q.y), 0.0) + length(max(q, 0.0)) - radius;
+}
+
+float corner_alpha(vec2 size, vec2 position, float radius_tl, float radius_tr, float radius_bl, float radius_br) {
 	vec2 relative_pos = (gl_FragCoord.xy - position);
 
-	vec2 top_left = abs(relative_pos - size) - size + radius;
-	vec2 top_right = abs(relative_pos - vec2(0, size.y)) - size + radius;
-	vec2 bottom_left = abs(relative_pos - vec2(size.x, 0)) - size + radius;
-	vec2 bottom_right = abs(relative_pos) - size + radius;
+	vec2 top_left = abs(relative_pos - size) - size + radius_tl;
+	vec2 top_right = abs(relative_pos - vec2(0, size.y)) - size + radius_tr;
+	vec2 bottom_left = abs(relative_pos - vec2(size.x, 0)) - size + radius_bl;
+	vec2 bottom_right = abs(relative_pos) - size + radius_br;
+
+	float dist = max(
+		max(get_dist(top_left, radius_tl), get_dist(top_right, radius_tr)),
+		max(get_dist(bottom_left, radius_bl), get_dist(bottom_right, radius_br))
```

Comment:
this can probably still be optimized considering we do the exact same operation on each value, the only val that changes is the `q` input, so we should be able to choose the proper q and then call `get_dist` once.